base/special/gamma.jl

```diff
@@ -496,20 +498,71 @@ for (f,T) in ((:f32,Float32),(:f16,Float16))
     end
 end
 
-zeta(s::Integer, z::Number) = zeta(Int(s), f64(z))
-zeta(s::Number, z::Number) = zeta(f64(s), f64(z))
-for f in (:digamma, :trigamma)
+
+function zeta(s::Integer, z::Number) 
+    x = float(z)
+    t = Int(s)
+    if typeof(x) === typeof(z) && typeof(t) === typeof(s)
+        throw MethodError(zeta,(s,t))
```

Comment:
Thanks.

It is a method error because this is a fallback definition.
If the most specific implementation of this  is `foo(::Number)`,
and if once you convert it to a `float`, the most specific implementation is *still* `foo(::Number)`,
because the type hasn't changed,
then we don't have a working implementation for this type.

This is the case for `trigamma(::BigFloat)`

I'll try and put that into a comment.
Should I write a block comment before all the fallback methods?
or write one per `throw`?